src/main/java/org/jenkins/plugins/lockableresources/queue/LockableResourcesQueueTaskDispatcher.java

```diff
@@ -75,8 +77,35 @@ public CauseOfBlockage canRun(Queue.Item item) {
 			if (item.task instanceof MatrixConfiguration) {
 			    MatrixConfiguration matrix = (MatrixConfiguration) item.task;
 			    params.putAll(matrix.getCombination());
+			} else {
```

Comment:
`else` does not seem to be correct there. Matrix projects deserve to get parameters as all other project types. Label calculation has higher priority (labels override parameters), so `MatrixConfiguration` check should go after the parameter injection